src/Message/AbstractRequest.php

```diff
@@ -929,7 +929,10 @@ public function sendData($data)
 
         if ($this->eventDispatcher) {
             // Log the request before it is sent.
-            $this->eventDispatcher->dispatch(Constants::OMNIPAY_REQUEST_BEFORE_SEND, new RequestEvent($this));
+            $this->eventDispatcher->dispatch(
+                Constants::OMNIPAY_REQUEST_BEFORE_SEND,
+                new RequestEvent($this, $action)
```

Comment:
this is gonna need more than just $action...some of the fetch requests have the same action (fetchByMerchantId) but are for different objects (autobills, transactions). you could include the object name. or you could include the request name using get_class

Reply:
Ohhh that's true. Thanks!

Reply:
Just logging the object name should suffice right? It doesn't seem like logging the action is necessary if it can be used by multiple objects.